### PR TITLE
chore: show release version in Grafana boards

### LINKED ISSUE #4525: Parameterize OSM Version in top-level Grafana JSON description field, so that upstream Grafana dashboards show compatible osm version.
As part of https://github.com/openservicemesh/osm/issues/4282, https://github.com/openservicemesh/osm/issues/4515 and https://github.com/openservicemesh/osm/issues/4516, upstream Grafana dashboards at https://grafana.com/orgs/openservicemesh/dashboards currently DO NOT show what the compatible underlying OSM version is.

This is because the top-level `description` field is not parameterized to show the OSM version.

https://github.com/openservicemesh/osm/blob/568e165ba3a0528eb3a490c98c7e3a4106f7fa42/charts/osm/grafana/dashboards/osm-mesh-envoy-details.json#L69

That top-level `description` field is what's shown at the top-level descriptor in our public dashboards repo:

image

image

Ideal description should be "Compatible with OSM v1.0.0", where the `v1.0.0` value is set by `.Chart.AppVersion`

**Scope (please mark with X where applicable)**
- New Functionality          [ ]
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- CI System                  [ ]
- Demo                       [ ]
- Project Release            [ ]

**Possible use cases**

### PR DESCRIPTION
**Pull request by johnsonshi**
Shows the latest osm release version in the description
field of osm's Grafana dashboard manifests. Also shows
a meaningful message describing each Grafana dashboard's
functionality within the same description field.

The description field is used by osm's Grafana.com
dashboard repo at
https://grafana.com/orgs/openservicemesh/dashboards

The description field is displayed prominently as the
sub-header title for each Grafana.com osm dashboard.
It is also the only location (for now) in upstream
Grafana.com to display compatible versions.

Contributes towards #4282.
Resolves #4525.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

Please describe how this change was tested. You could include supporting information
such as logs, snippets, and screenshots.

**Testing done**: Ran the bookstore demo with Grafana enabled, port-forwarded Grafana, and inspected the validity of Grafana data.

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [X] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change? No

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)? Not yet. Will create separate PR in osm-docs.

### CODE REVIEW

File: charts/osm/grafana/dashboards/osm-control-plane.json
Comment (snehachhabria) [inline charts/osm/grafana/dashboards/osm-control-plane.json:64]: my concert with specifying the version here adds an overhead of updating this every time there's a new release

Comment (jaellio) [reply charts/osm/grafana/dashboards/osm-control-plane.json:64]: Could the version be an input variable like DS_PROMETHEUS or CONTROL_PLANE_NAMESPACE?

Comment (jaellio) [reply charts/osm/grafana/dashboards/osm-control-plane.json:64]: I misspoke in my previous comment. I am in favor of updating the **description** to include the OSM version.

Also, an input variable would not make sense for the OSM version. Perhaps something like with would work https://grafana.com/docs/grafana/latest/administration/configuration/#env-provider

File: charts/osm/grafana/dashboards/osm-data-plane-performance.json
Comment (jaellio) [inline charts/osm/grafana/dashboards/osm-data-plane-performance.json:64]: nit: Since the description here is used as the revision description, is it possible to specify information like "This dashboard lets you view the performance of OSM’s data plane" somewhere else? Otherwise, this information might make the revision list a bit cluttered.

**Review by jaellio — APPROVED** [review]

File: charts/osm/grafana/dashboards/osm-data-plane-performance.json
Comment (johnsonshi) [reply charts/osm/grafana/dashboards/osm-data-plane-performance.json:64]: Unfortunately not. In fact, the Grafana dashboard JSON model (documented [here](https://grafana.com/docs/grafana/latest/dashboards/json-model/)) does not even have the `description` field in its JSON. In fact, only the `title` field is documented in the Grafana dashboard JSON model as per Grafana docs (I've searched everywhere and there doesn't seem to be documentation on even the `description` field).

This is how it looks like. The `title` json field corresponds to the title highlighted below, while the `description` field corresponds to the "short description" field highlighted below.

image

I think this PR is good to go and could be merged. @jaellio - as I don't have maintainer permissions, can you merge it for me?

Comment (jaellio) [reply charts/osm/grafana/dashboards/osm-data-plane-performance.json:64]: I cannot merge until there is another review. @snehachhabria would you mind taking another look at this PR?

**Review by snehachhabria — APPROVED** [review]

File: charts/osm/grafana/dashboards/osm-data-plane-performance.json
Comment (johnsonshi) [reply charts/osm/grafana/dashboards/osm-data-plane-performance.json:64]: Ok @jaellio this can now be merged.

**Pull request merged**
